src/legendmeta/core.py

```diff
@@ -136,12 +139,35 @@ def channelmap(
         >>> channel.analysis.usability
         'on'
 
+        >>> channel = lmeta.channelmap(on=datetime.now(), system='cal').V05267B
+
+        >>> channel = lmeta.channelmap('p08 r005', system='cal').V05267B
+
         See Also
         --------
         .textdb.TextDB.on
         """
         if on is None:
             on = datetime.now()
+            if system is None:
+                system = "all"
+
+        if system is None:
+            msg = "Use of channelmap with an on arg but no system arg is deprecated and will raise an error in the future. Provide a value (e.g. 'all', 'cal', 'phy', etc.)"
+            log.warning(msg)
+            # msg = "System cannot be None. Provide a value (e.g. 'all', 'cal', 'phy', etc.)"
+            # raise ValueError(msg)
+
+        try:
+            pd, run = re.fullmatch("([pP][0-9]+)\\s*([rR][0-9]+)", on).group(1, 2)
+            runinfo = self.dataprod.runinfo[pd][run]
```

Comment:
Will this generate an understandable error message if the string does not match?